server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java

```diff
@@ -0,0 +1,303 @@
+        .setLong(7, webhook.createdAt())
+        .execute()
+        .commit();
+    } else {
+      LOGGER.info("Unable to migrate inconsistent webhook (entry deleted from PROPERTIES) : " + webhook);
+    }
+
+    return webhook.rowIds();
+  }
+
+  private String deleteRequest(List<Long> toBeDeleted) {
+    return new StringBuilder()
+      .append("DELETE from properties WHERE id in (")
+      .append(Joiner.on(',').join(toBeDeleted))
+      .append(")")
+      .toString();
+  }
+
+  private List<Webhook> extractGlobalWebhooksFrom(Context context, Map<String, PropertyRow> properties, String[] values) throws SQLException {
+    String defaultOrganizationUuid = defaultOrganizationUuidProvider.get(context);
+    return Arrays.stream(values)
+      .map(value -> new Webhook(
+        properties.get("sonar.webhooks.global." + value + ".name"),
+        properties.get("sonar.webhooks.global." + value + ".url"),
+        defaultOrganizationUuid, null))
+      .collect(toList());
+  }
+
+  private List<Webhook> extractProjectWebhooksFrom(Map<String, PropertyRow> properties, String[] values, Map<Long, String> projectsKeys) {
+    return Arrays.stream(values)
+      .map(value -> new Webhook(
+        properties.get("sonar.webhooks.project." + value + ".name"),
+        properties.get("sonar.webhooks.project." + value + ".url"),
+        null, projectsKeys))
+      .collect(toList());
+  }
+
+  private class PropertyRow {
+
+    private final Long id;
+    private final String key;
+    private final Long resourceId;
+    private final String value;
+    private final Long createdAt;
+
+    public PropertyRow(long id, String key, Long resourceId, String value, Long createdAt) {
+      this.id = id;
+      this.key = key;
+      this.resourceId = resourceId;
+      this.value = value;
+      this.createdAt = createdAt;
+    }
+
+    public Long id() {
+      return id;
+    }
+
+    public String key() {
+      return key;
+    }
+
+    public Long resourceId() {
+      return resourceId;
+    }
+
+    public String value() {
+      return value;
+    }
+
+    public Long createdAt() {
+      return createdAt;
+    }
+
+    @Override
+    public String toString() {
+      return "{" +
+        "id=" + id +
+        ", key='" + key + '\'' +
+        ", resourceId=" + resourceId +
+        ", value='" + value + '\'' +
+        ", createdAt=" + createdAt +
+        '}';
+    }
+  }
+
+  private class Webhook {
+
+    private final PropertyRow name;
+    private final PropertyRow url;
+    private String organisationUuid;
+    private String projectUuid;
+
+    public Webhook(@Nullable PropertyRow name, @Nullable PropertyRow url, @Nullable String organisationUuid, @Nullable Map<Long, String> projectsKeys) {
+      this.name = name;
+      this.url = url;
+      this.organisationUuid = organisationUuid;
+      if (StringUtils.isBlank(organisationUuid)) {
+        this.projectUuid = projectsKeys.get(name.resourceId());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "projectsKeys" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWGPwo1FFMFpNJFuf2rm&open=AWGPwo1FFMFpNJFuf2rm)

[ISSUE_KEY]: AWGPwo1FFMFpNJFuf2rm